include/nil/crypto3/zk/snark/systems/plonk/placeholder/prover.hpp

```diff
@@ -193,67 +193,35 @@ namespace nil {
-                        _challenge_point = {_proof.eval_proof.challenge};
+                        _proof.eval_proof.challenge = transcript.template challenge<FieldType>();
+                        //_proof.eval_proof.challenge = typename FieldType::value_type(7);
 
-                        std::vector<std::vector<typename FieldType::value_type>> evaluation_points_quotient = {_challenge_point};
-                        _combined_poly[2].insert(
-                            std::end(_combined_poly[2]),
-                            std::make_move_iterator(std::begin(T_splitted_dfs)),
-                            std::make_move_iterator(std::end(T_splitted_dfs)));
+                        generate_evaluation_points();
 
-                        // public
-                        _proof.eval_proof.combined_value = compute_combined_value(
-                            std::move(variable_values_evaluation_points),
-                            std::move(evaluation_points_quotient),
-                            std::move(variable_values_precommitment),
-                            std::move(T_precommitment),
-                            std::move(evaluation_points_v_p),
-                            std::move(permutation_argument.permutation_poly_precommitment));
+                        _proof.eval_proof.eval_proof = _commitment_scheme.proof_eval(transcript);                        
 
-                        _proof.fixed_values_commitment = preprocessed_public_data.common_data.commitments.fixed_values;
-                        return std::move(_proof);
+                        return _proof;
                     }
 
                 private:
-                    typename commitment_scheme_type::precommitment_type precommit_witness() {
-                        PROFILE_PLACEHOLDER_SCOPE("witness_precommit_time");
-                        return algorithms::precommit<commitment_scheme_type>(
-                            _combined_poly[0], fri_params.D[0], fri_params.step_list.front());
-                    }
-
                     std::vector<polynomial_dfs_type> quotient_polynomial_split_dfs() {
-                        std::vector<polynomial_type> T_splitted = 
-                            detail::split_polynomial<FieldType>(
-                                quotient_polynomial(), fri_params.max_degree);
+                        // TODO: pass max_degree parameter placeholder
+                        std::vector<polynomial_type> T_splitted = detail::split_polynomial<FieldType>( 
+                            quotient_polynomial(), table_description.rows_amount - 1
+                        );
 
                         PROFILE_PLACEHOLDER_SCOPE("split_polynomial_dfs_conversion_time");
 
-                        std::vector<polynomial_dfs_type> T_splitted_dfs(
-                            T_splitted.size(), polynomial_dfs_type(0, fri_params.D[0]->size()));
+                        std::vector<polynomial_dfs_type> T_splitted_dfs;

```

Comment:
You can pass the size to the constructor, like    std::vector<polynomial_dfs_type> T_splitted_dfs(T_splitted.size());